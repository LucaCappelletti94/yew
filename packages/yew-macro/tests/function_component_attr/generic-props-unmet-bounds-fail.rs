use yew::prelude::*;

#[derive(Clone, Properties, PartialEq)]
struct Props {
    a: usize,
}

#[function_component(Comp)]
fn comp<P>(_props: &P) -> Html
where
    P: Properties + PartialEq,
{
    html! {
        <p></p>
    }
}

struct MissingTypeBounds;

fn compile_fail_unmet_bounds() {
    // parameter doesn't match bounds
    html! { <Comp<MissingTypeBounds> /> };
}

fn main() {}
