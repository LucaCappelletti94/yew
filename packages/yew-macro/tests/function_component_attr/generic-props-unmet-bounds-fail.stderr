error[E0277]: the trait bound `Comp<MissingTypeBounds>: yew::BaseComponent` is not satisfied
  --> tests/function_component_attr/generic-props-unmet-bounds-fail.rs:22:14
   |
22 |     html! { <Comp<MissingTypeBounds> /> };
   |              ^^^^^^^^^^^^^^^^^^^^^^^ the trait `yew::BaseComponent` is not implemented for `Comp<MissingTypeBounds>`
   |
   = help: the trait `yew::BaseComponent` is implemented for `Comp<P>`

error[E0277]: the trait bound `MissingTypeBounds: yew::Properties` is not satisfied
  --> tests/function_component_attr/generic-props-unmet-bounds-fail.rs:22:14
   |
22 |     html! { <Comp<MissingTypeBounds> /> };
   |              ^^^^ the trait `yew::Properties` is not implemented for `MissingTypeBounds`
   |
   = help: the following other types implement trait `yew::Properties`:
             bool
             char
             isize
             i8
             i16
             i32
             i64
             i128
           and $N others
note: required by a bound in `Comp`
  --> tests/function_component_attr/generic-props-unmet-bounds-fail.rs:11:8
   |
8  | #[function_component(Comp)]
   |                      ---- required by a bound in this struct
...
11 |     P: Properties + PartialEq,
   |        ^^^^^^^^^^ required by this bound in `Comp`
   = note: this error originates in the macro `html` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: the function or associated item `new` exists for struct `VChild<Comp<MissingTypeBounds>>`, but its trait bounds were not satisfied
  --> tests/function_component_attr/generic-props-unmet-bounds-fail.rs:22:14
   |
8  | #[function_component(Comp)]
   | ------------------------- doesn't satisfy `Comp<MissingTypeBounds>: yew::BaseComponent`
...
22 |     html! { <Comp<MissingTypeBounds> /> };
   |              ^^^^ function or associated item cannot be called on `VChild<Comp<MissingTypeBounds>>` due to unsatisfied trait bounds
   |
   = note: the following trait bounds were not satisfied:
           `Comp<MissingTypeBounds>: yew::BaseComponent`
note: the trait `yew::BaseComponent` must be implemented
  --> $WORKSPACE/packages/yew/src/html/component/mod.rs
   |
   | pub trait BaseComponent: Sized + 'static {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in the macro `html` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `Comp<MissingTypeBounds>: yew::BaseComponent` is not satisfied
  --> tests/function_component_attr/generic-props-unmet-bounds-fail.rs:22:14
   |
22 |     html! { <Comp<MissingTypeBounds> /> };
   |              ^^^^ the trait `yew::BaseComponent` is not implemented for `Comp<MissingTypeBounds>`
   |
   = help: the trait `yew::BaseComponent` is implemented for `Comp<P>`
note: required by a bound in `VChild`
  --> $WORKSPACE/packages/yew/src/virtual_dom/vcomp.rs
   |
   | pub struct VChild<COMP: BaseComponent> {
   |                         ^^^^^^^^^^^^^ required by this bound in `VChild`
   = note: this error originates in the macro `html` (in Nightly builds, run with -Z macro-backtrace for more info)
